From be738bc7a7cfbcbf96b42666dc621cb948e6e828 Mon Sep 17 00:00:00 2001
From: "R, Kannappan" <r.kannappan@intel.com>
Date: Fri, 17 Oct 2014 13:24:20 +0530
Subject: [PATCH 1/1] em: fix build errors when power supply configs are not
 defined

fix the build errors when POWER_SUPPLY_CHARGER/BATTID
is not defined.

Change-Id: I1aa3277588b30b1a533c62bdc71e8eb0980a28d4
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
---
 drivers/extcon/extcon-tsu6111.c        |    2 +-
 drivers/mfd/intel_soc_pmic_dc.c        |    4 ++--
 drivers/platform/x86/intel_em_config.c |    8 +++++++-
 include/linux/extcon/extcon-tsu6111.h  |    6 +++---
 4 files changed, 13 insertions(+), 7 deletions(-)

diff --git a/drivers/extcon/extcon-tsu6111.c b/drivers/extcon/extcon-tsu6111.c
index 84bd5b3f47f6..3ace083a59f3 100644
--- a/drivers/extcon/extcon-tsu6111.c
+++ b/drivers/extcon/extcon-tsu6111.c
@@ -662,7 +662,7 @@ static struct tsu6111_pdata tsu_drvdata = {
 	.disable_vbus = bq24192_vbus_disable,
 	.is_vbus_online = dc_ti_vbus_on_status,
 #else
-	.enable_vbus = dummy_vubs_enable,
+	.enable_vbus = dummy_vbus_enable,
 	.disable_vbus = dummy_vbus_disable,
 	.is_vbus_online = dummy_vbus_status,
 #endif
diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index 0b79317e6426..272f4050bd1a 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -292,14 +292,14 @@ struct intel_pmic_irqregmap dollar_cove_irqregmap[] = {
 };
 
 
+static struct ps_batt_chg_prof ps_batt_chrg_prof;
+static struct ps_pse_mod_prof *pse_mod_prof;
 #ifdef CONFIG_POWER_SUPPLY_CHARGER
 
 #define DC_CHRG_CHRG_CUR_NOLIMIT	1800
 #define DC_CHRG_CHRG_CUR_MEDIUM		1400
 #define DC_CHRG_CHRG_CUR_LOW		1000
 
-static struct ps_batt_chg_prof ps_batt_chrg_prof;
-static struct ps_pse_mod_prof *pse_mod_prof;
 static struct power_supply_throttle dc_chrg_throttle_states[] = {
 	{
 		.throttle_action = PSY_THROTTLE_CC_LIMIT,
diff --git a/drivers/platform/x86/intel_em_config.c b/drivers/platform/x86/intel_em_config.c
index f81546db2f3c..3d23d669cea9 100644
--- a/drivers/platform/x86/intel_em_config.c
+++ b/drivers/platform/x86/intel_em_config.c
@@ -15,7 +15,9 @@
 #define EM_CONFIG_OEM0_NAME "OEM0"
 #define EM_CONFIG_OEM1_NAME "OEM1"
 
+#ifdef DEBUG
 static void dump_chrg_profile(const struct ps_pse_mod_prof *chrg_prof);
+#endif
 
 
 static struct ps_pse_mod_prof chrg_prof;
@@ -32,7 +34,7 @@ static int em_config_get_acpi_table(char *name, void *data, int data_size)
 	status = acpi_get_table_with_size(name , 0,
 					&acpi_tbl, &tbl_size);
 	if (ACPI_SUCCESS(status)) {
-		pr_info("EM:%s  table found, size=%d\n", name, tbl_size);
+		pr_info("EM:%s  table found, size=%d\n", name, (int)tbl_size);
 		if (tbl_size < (data_size + hdr_size)) {
 			pr_err("EM:%s table incomplete!!\n", name);
 		} else {
@@ -125,14 +127,18 @@ static int __init em_config_init(void)
 
 	batt_chrg_prof.batt_prof = &chrg_prof;
 
+#ifdef CONFIG_POWER_SUPPLY_BATTID
 	battery_prop_changed(POWER_SUPPLY_BATTERY_INSERTED, &batt_chrg_prof);
+#endif
 	return 0;
 }
 early_initcall(em_config_init);
 
 static void __exit em_config_exit(void)
 {
+#ifdef CONFIG_POWER_SUPPLY_BATTID
 	batt_chrg_prof.chrg_prof_type = CHRG_PROF_NONE;
 	battery_prop_changed(POWER_SUPPLY_BATTERY_INSERTED, &batt_chrg_prof);
+#endif
 }
 module_exit(em_config_exit);
diff --git a/include/linux/extcon/extcon-tsu6111.h b/include/linux/extcon/extcon-tsu6111.h
index 71b0e6778fb7..bcf5521d7f2e 100644
--- a/include/linux/extcon/extcon-tsu6111.h
+++ b/include/linux/extcon/extcon-tsu6111.h
@@ -26,9 +26,9 @@ struct tsu6111_pdata {
 	bool charging_compliance_override;
 };
 
-inline int dummy_vubs_enable(void) { return -1; }
-inline int dummy_vubs_disable(void) { return -1; }
-inline int dummy_vubs_status(void) { return -1; }
+inline int dummy_vbus_enable(void) { return -1; }
+inline int dummy_vbus_disable(void) { return -1; }
+inline int dummy_vbus_status(void) { return -1; }
 
 extern int dc_ti_vbus_on_status(void);
 
-- 
1.7.9.5

